### DISCUSSION TITLE
UserWarning: FP16 is not supported on CPU ?

### DISCUSSION THREAD

**Discussion by Dimio12** [openai/whisper#1194] [Q&A]
Hi everyone!

How can i change FP16 to FP32?

I have this error

C:\Users\DIMIO\AppData\Local\Programs\Python\Python39\lib\site-packages\whisper\transcribe.py:114: UserWarning: FP16 is not supported on CPU; using FP32 instead
warnings.warn("FP16 is not supported on CPU; using FP32 instead")

**Comment by glangford** [comment]
On the command line give the option `--fp16 False`.

**Reply by Dimio12** [reply]
Just like this in the windown Command Prompt?

C:\Users\Dimio>whisper --fp16 False

Sorry im new on this territorys

**Reply by glangford** [reply]
`whisper myaudio.mp3 --fp16 False`

And for the other options available from the command prompt, enter
`whisper --help`

In particular see the options `--task` and `--output_format` to get started.

**Reply by noises-modular** [reply]
Is there an equivalent way to set that value to false in python?

**Reply by phineas-pta** [reply]
`transcribe(..., fp16=True)`

**Comment by tianyi725** [comment]
hi everyone, I had the same error.

I added --fp16 False

Still have a problem, do not know how to modify?


whisper h:\M.mp4 --model tiny --fp16 False
E:\Python\lib\site-packages\whisper\timing.py:57: NumbaDeprecationWarning: The 'nopython' keyword argument was not supplied to the 'numba.jit' decorator. The implicit default value for this argument is currently False, but it will be changed to True in Numba 0.59.0. See https://numba.readthedocs.io/en/stable/reference/deprecation.html#deprecation-of-object-mode-fall-back-behaviour-when-using-jit for details.

**Reply by phineas-pta** [reply]
what you got is a warning, not error

**Reply by tianyi725** [reply]
> what you got is a warning, not error
All information
Why is it always FileNotFoundError？
thanks

whisper h:\M.mp4 --fp16 False --model tiny
E:\Python\lib\site-packages\whisper\timing.py:57: NumbaDeprecationWarning: The 'nopython' keyword argument was not supplied to the 'numba.jit' decorator. The implicit default value for this argument is currently False, but it will be changed to True in Numba 0.59.0. See https://numba.readthedocs.io/en/stable/reference/deprecation.html#deprecation-of-object-mode-fall-back-behaviour-when-using-jit for details.
  def backtrace(trace: np.ndarray):
Traceback (most recent call last):
  File "E:\Python\lib\runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "E:\Python\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "E:\Python\Scripts\whisper.exe\__main__.py", line 7, in <module>
  File "E:\Python\lib\site-packages\whisper\transcribe.py", line 437, in cli
    result = transcribe(model, audio_path, temperature=temperature, **args)
  File "E:\Python\lib\site-packages\whisper\transcribe.py", line 121, in transcribe
    mel = log_mel_spectrogram(audio, padding=N_SAMPLES)
  File "E:\Python\lib\site-packages\whisper\audio.py", line 130, in log_mel_spectrogram
    audio = load_audio(audio)
  File "E:\Python\lib\site-packages\whisper\audio.py", line 46, in load_audio
    ffmpeg.input(file, threads=0)
  File "E:\Python\lib\site-packages\ffmpeg\_run.py", line 313, in run
    process = run_async(
  File "E:\Python\lib\site-packages\ffmpeg\_run.py", line 284, in run_async
    return subprocess.Popen(
  File "E:\Python\lib\subprocess.py", line 947, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "E:\Python\lib\subprocess.py", line 1416, in _execute_child
    hp, ht, pid, tid = _winapi.CreateProcess(executable, args,
FileNotFoundError: [WinError 2] 系统找不到指定的文件。

**Reply by phineas-pta** [reply]
error because you don't have ffmpeg

**Comment by Mansi9991** [comment]


**Reply by phineas-pta** [reply]
i'm not part of the openai team, also there're other people here to help you, you don't have to tag me

**Reply by Mansi9991** [reply]
Ok I am sorry

**Comment by WittyF0X** [comment]
> Ok I am sorry

Hi friend, you need to install this package
`pip install setuptools-rust`
Read the documentation
[https://github.com/openai/whisper](https://github.com/openai/whisper)
And you can reinstall everything
Good luck
